src/Features/OnboardingTasks.php

```diff
@@ -716,6 +717,57 @@ function( $has_installed, $bundle ) {
 		return apply_filters( 'woocommerce_admin_onboarding_tasks', $task_lists );
 	}
 
+	/**
+	 * Retrieve a task list by ID.
+	 *
+	 * @param String $task_list_id Task list ID.
+	 *
+	 * @return Object
+	 */
+	public static function get_task_list_by_id( $task_list_id ) {
+		foreach ( self::get_task_lists() as $task_list ) {
+			if ( $task_list['id'] === $task_list_id ) {
+				return $task_list;
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Retrieve single task.
+	 *
+	 * @param String $task_id Task ID.
+	 * @param String $task_list_id Task list ID.
+	 *
+	 * @return Object
+	 */
+	public static function get_task_by_id( $task_id, $task_list_id = null ) {
+
+		$task_list = $task_list_id ? self::get_task_list_by_id( $task_list_id ) : null;
+
+		if ( $task_list_id && ! $task_list ) {
+			return null;
+		}
+
+		$tasks_to_search = $task_list_id ? $task_list['tasks'] : array_reduce(
```

Comment:
Not a blocker to getting this merged, especially since we may update the task API by the end of cycle, but just a quick note that this loops over the tasks twice.

This most likely will not have any meaningful impact on performance given how small the task list is, but I'd opt for a single loop where possible if we update this.

Reply:
I don't believe this is really any different than the nested `foreach` loops we'd be using otherwise? The reduce only loops through the task lists (only 1 currently), and then the following `foreach` loop just loops through the relevant tasks. The benefit is that if `$task_list_id` is provided, you'll skip the first loop entirely and only loop through the tasks for the specified list. 

This depends a bit on the performance of `array_merge`; is that what you're referring to? Not too familiar with the inner-workings there.

Reply:
Yes and no.  You're correct if we simply nest the foreach and let it run.  However, with the nested option we can return early or break the loop which has the ability to be more performant.

But again, this is a very nitpicky comment and I'm hoping we can create a better API around registering/getting tasks in the backend that would make this obsolete anyway.

Reply:
Yes and no.  You're correct if we simply nest the foreach and let it run.  However, with the nested option we can return early or break the loop which has the ability to be more performant.

But again, this is a very nitpicky comment and I'm hoping we can create a better API around registering/getting tasks in the backend that would make this obsolete anyway.

Reply:
True, although the reduce is only looping through the list objects, and the subsequent `foreach` loop that goes through the tasks does return early as you mention. Hopefully all moot as we improve this, but I was just curious if I was missing something. 🙏